user/nix/files/nix.profile.d

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+if [ -z "$__NIX_SET_ENV_DONE" ]; then
+	. /etc/default/nix
+
+	# default paths
+	DATA_PATH="${XDG_DATA_DIRS-/usr/local/share:/usr/share}"
+	XDG_STATE_HOME="${XDG_STATE_HOME-$HOME/.local/state}"
```

Comment:
Another nit pick. Shouldn't the expansion usually be `:-` instead of `-`, so the variable is checked for both its existence and whether its value is null?